exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillPreparedStatementImpl.java

```diff
@@ -259,4 +261,13 @@ public void setObject(int parameterIndex, Object x, SQLType targetSqlType) throw
     checkOpen();
     super.setObject(parameterIndex, x, targetSqlType);
   }
+
+  @Override
+  public void setLargeMaxRows(long maxRowCount) throws SQLException {
+    super.setLargeMaxRows(maxRowCount);
+    try (Statement statement = this.connection.createStatement()) {
+      statement.execute("ALTER SESSION SET `" + ExecConstants.QUERY_MAX_ROWS + "`=" + maxRowCount);
+      statement.close();
```

Comment:
Please remove this line, `statement` will be closed automatically since it is used in `try with resources`.

Reply:
👍